src/test/java/rx/SingleTest.java

```diff
@@ -1353,4 +1386,364 @@ public String call() throws Exception {
         int numberOfErrors = retryCounter.getOnErrorEvents().size();
         assertEquals(retryCount, numberOfErrors);
     }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testTakeUntilSuccess() {
```

Comment:
For the test prefix: 322ae576172d2c55bbaaf258314460e396cd0d55

I wasn't sure what to do about the names. I figured since it was in `SingleTest` that I could assume any test without an explicit "Observable" mention was using "Single". Maybe it's not that obvious though. @akarnokd what do you think?